### PR TITLE
Bump kubemacpool v0.11.0

### PR DESCRIPTION
**Pull request by ormergi**
This PR continues #377 kubemacpool bumping:

> This PR bump kubemacpool to version v0.11.0, to introduce opt-in per namespace functionality, cert-manager upgrade so that kubemacpool can run on openshift, fixing flaky tests, and other infra updates.
> 
> kubemacpool bumped to v0.11.0

```release-note
Bump kubemacpool v0.11.0
```

### CODE REVIEW

**Comment by phoracek** [comment]
There is an alarming failure in e2e logs:

```
           Warning  Unhealthy  9m46s (x9 over 14m)   kubelet, node01    Readiness probe failed: Get https://10.244.0.38:8000/readyz: dial tcp 10.244.0.38:8000: connect: connection refused
          Warning  BackOff    4m46s (x37 over 14m)  kubelet, node01    Back-off restarting failed container 
```

**Comment by RamLavi** [comment]
>There is an alarming failure in e2e logs:

@phoracek this is by design in the non-leader kmp pod, (since it's controller-runtime manage hasn't been started, then the readiness probe fails)
need to make sure that this is not an issue of the leader pod (where it should NOT happen)
also make sure if this make us fail the tests..

**Comment by phoracek** [comment]
@ormergi even if this proves to be flaky, please consider figuring out what is wrong instead of `/retest`. We don't want to bring any hidden failures to the code.

File: pkg/apply/merge.go [hunk 1]
```diff
@@ -178,20 +178,23 @@ func mergeWebhookConfiguration(current, updated *unstructured.Unstructured) erro
 			continue
 		}
```
Comment (RamLavi) [inline pkg/apply/merge.go:180]: is there a reason to keep this commit in this PR?

File: pkg/apply/merge_test.go [hunk 2]
```diff
@@ -374,20 +337,131 @@ webhooks:
...
+			err := apply.MergeMetadataForUpdate(current, updated)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(updated.GetCreationTimestamp()).To(Equal(current.GetCreationTimestamp()))
+			Expect(updated.GetGeneration()).To(Equal(current.GetGeneration()))
+			Expect(updated.GetResourceVersion()).To(Equal(current.GetResourceVersion()))
+			Expect(updated.GetSelfLink()).To(Equal(current.GetSelfLink()))
+			Expect(updated.GetUID()).To(Equal(current.GetUID()))
 		})
 	})
+
```
Comment (RamLavi) [inline pkg/apply/merge_test.go:466]: please remove line

File: pkg/apply/merge.go (hunk 1 above)
Comment (phoracek) [reply pkg/apply/merge.go:180]: +1, please add it to a separate one. The PR is huge as it is.

**Comment by phoracek** [comment]
/hold

Should be split.

**Comment by RamLavi** [comment]
/unhold
split was performed

**Comment by RamLavi** [comment]
/hold
should wait for #391 and #390 in order to fix issues found during integration.

**Comment by ormergi** [comment]
Closed in favor #395

**Pull request closed without merging**